### DISCUSSION TITLE
How to deploy to Replit?

### DISCUSSION THREAD

**Discussion by Mennaruuk** [searxng/searxng#976] [Q&A]
Replit is a great service, they have a free plan that allows users to self host. Of course, there are drawbacks. The REPLs are publicly visible, so if you want something to stay private, you have to upgrade. [Server usage is monitored for abuse](https://replit.com/site/terms):

> Replit has the right but not the obligation to monitor, edit, remove, or share User Content provided by users where it believes in good faith that it is necessary to comply with applicable law or to ensure the consistency, security, and stability of our Service.

However, using Replit has the advantage of the user hosting searxng for themselves and making modifications to their liking, respecting the license. Plus, the project is going to basically be a one-click installation process, [check out Libreddit's example](https://github.com/spikecodes/libreddit#5-replitherokuglitch), they have three buttons for Replit, Heroku, and Glitch, and users clicking these buttons are taken instantly to a clone of Libreddit waiting to be self hosted.

![image](https://user-images.githubusercontent.com/52135169/157159259-8b63d1e0-00e5-4900-8cb6-4b832f47a433.png)

I tried getting it to work, but I haven't been successful. I tried choosing Python 3 as the default language and running `make install`, but that did nothing. I also tried choosing Node.js as the language and running `npm start` (that's what Replit gives you as the default choices), but that didn't do much. Also, I attempted deploying the Docker image, but [Replit struggled to install Docker](https://replit.com/talk/ask/install-pkg-docker/39255/177830).

Your help is greatly appreciated! Keep searxng :)

**Comment by Linerly** [comment]
I have tried hosting SearXNG on Replit a while ago

Go to https://replit.com/github/searxng/searxng/, and wait for a while

Once you get to the “Detecting Language” phase, refresh the page, and it should use Bash as the language, run SearXNG by creating a `.replit` file and add these into the file:
```
language="bash"
run="pip install -r requirements.txt && pip install -e . && ./searx/webapp.py"
```

After that, configure your SearXNG instance, at least the secret token and address (set it to 0.0.0.0, no need to change the port)

…click on Run and you should be able to use SearXNG

**Reply by digitalknk** [reply]
This does run searxng but doesn't expose it, as far as I can see there is no url that replit will provide you to actually be able to use it.

**Reply by Linerly** [reply]
Oh, also be sure to set the address to 0.0.0.0

**Reply by Linerly** [reply]
I recently messed up my Repl where my instance of SearXNG is hosted, consisting of errors installing dependencies…

**Reply by Linerly** [reply]
I almost forgot that Replit now uses Nix

Being unfamilar with Nix though, it pretty much breaks most of my existing Repls
